x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml

```diff
@@ -14,6 +14,10 @@ template:
         type: passthrough
         dynamic: true
         priority: 10
+      # workaround for https://github.com/elastic/elasticsearch/issues/99123
+      _metric_names_hash:
+        type: keyword
```

Comment:
Q: will a number be more lightweight, as you're using a 8 digit hex anyway?

Reply:
At the moment, numbers can't leverage run-length encoding. So it's actually lighter to use a keyword here as all dimensions are incorporated into the _tsid, which we sort by. Therefore, all values for the same tsid are equal and can be compressed very efficiently.